projects/astral-accessibility/src/lib/controls/contrast.component.ts

```diff
@@ -82,23 +83,30 @@ import { AstralCheckmarkSvgComponent } from "../util/astral-checksvg.component";
   `,
   imports: [NgIf, NgClass, AstralCheckmarkSvgComponent],
 })
-export class ContrastComponent {
+export class ContrastComponent extends AccessibilityComponent {
+  constructor() {
+    super();
+    this.currentState = super.setLogic("astralAccessibility_contrastState");
+  }
   document = inject(DOCUMENT);
 
-  currentState = 0;
+  currentState = super.getState("astralAccessibility_contrastState");
   base = "Contrast";
   states = [this.base, "Invert", "High Contrast", "Dark High Contrast"];
 
   _style: HTMLStyleElement;
 
   nextState() {
-    this.currentState += 1;
-    this.currentState = this.currentState % 4;
+    this.currentState = super.changeState(
+      this.currentState,
+      "astralAccessibility_contrastState",
+      this.states.length,
```

Comment:
I feel like first and last argument can be defined only in `accessibility.component.ts`, and we can make the second argument be defined as a class variable and have the parent read it as well. Hope that makes sense.

Read more on Template Method Design Pattern: https://refactoring.guru/design-patterns/template-method